packages/react-app-revamp/components/_pages/WizardFormCreateContest/Step2/useDeployToken.ts

```diff
@@ -57,20 +69,30 @@ export function useDeployToken(form: any) {
         hash: contract.deployTransaction.hash,
       });
       stateContractDeployment.setIsSuccess(true);
-      setDeployTokenData({
-        hash: receipt.transactionHash,
-        address: contract.address,
-      });
-      if (modalDeployTokenOpen === false)
+      if (isDeployingSubmissionToken === true) {
+        setDeploySubmissionTokenData({
+          hash: receipt.transactionHash,
+          address: contract.address,
+        });
+      } else {
+        setDeployTokenData({
+          hash: receipt.transactionHash,
+          address: contract.address,
+        });
+      }
+      if (modalDeployTokenOpen === false && modalDeploySubmissionTokenOpen === false) {
         toast.success(
           `The contract for your token ${values.tokenName} ($${values.tokenSymbol}) was deployed successfully!`,
         );
-
+      }
       stateContractDeployment.setIsLoading(false);
+      stateContractDeployment.setIsSuccess(true);
+      stateContractDeployment.setIsError(false);
+      stateContractDeployment.setErrorMessage(null);
       form.reset();
     } catch (e) {
       console.error(e);
-      if (modalDeployTokenOpen === false)
+      if (modalDeployTokenOpen === false && modalDeploySubmissionTokenOpen === false)
```

Comment:
Why do we only want to show an error when deploying voting tokens and not submission tokens?

Reply:
it's to display a toast with the error. It looks weird to display a toast on top of a modal, so we only display the toast if there's no modal open

Reply:
Gotcha! Is there any way to do error reporting if the modal is open so the user knows what went wrong?